### PR TITLE
Add rubygem-google-protobuf package

### PR DESCRIPTION
**Pull request by stejskalleos**
For https://github.com/theforeman/foreman-packaging/pull/8248

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Please add this to `plugins`, not `foreman`. So
```
./add_gem_package.sh google-protobuf default plugins foreman-plugins-el8
```

This is because the package should end up in the plugins repository, not the main repository.

**Comment by stejskalleos** [comment]
> Please add this to `plugins`, not `foreman`. So
> 
> ```
> ./add_gem_package.sh google-protobuf default plugins foreman-plugins-el8
> ```
> 
> This is because the package should end up in the plugins repository, not the main repository.

Should I do that also with packages in https://github.com/theforeman/foreman-packaging/pull/8248? Right now they are in `packages/foreman`, not in `packages/plugins` folder

**Comment by stejskalleos** [comment]
Linter is failing on `msg: Could not find specfile` even when it's there.

@theforeman/packaging What are proper steps for adding new package?
Looks like running `./add_gem_package.sh google-protobuf default foreman-plugins-el8` is not enough and additional steps must be done, just don't know where to look for more information

File: package_manifest.yaml [hunk 1]
```diff
@@ -194,6 +194,7 @@ foreman_core_packages:
     rubygem-google-apis-sqladmin_v1beta4: {}
     rubygem-google-apis-storage_v1: {}
     rubygem-google-cloud-env: {}
+    rubygem-google-protobuf: {}
```
Comment (ehelms) [inline package_manifest.yaml:197]: This needs to be moved under the plugins section around like 674/675

File: packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+%{gem_libdir}
+%exclude %{gem_cache}
+%{gem_spec}
+
+%files doc
+%doc %{gem_docdir}
+%{gem_instdir}/tests
+
+%changelog
+* Tue Sep 27 2022 Leos Stejskal <lstejska@redhat.com> 3.21.6-1
```
Comment (ehelms) [inline packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec:81]:
```suggestion
* Tue Sep 27 2022 Leos Stejskal <lstejska@redhat.com> 1:3.21.6-1
```

**Comment by stejskalleos** [comment]
rpm job is failing on error:
```
package: rubygem-google-protobuf-doc-1:3.21.6-1.el8.noarch from downloaded_rpms
unresolved deps:
rubygem-google-protobuf = 3.21.6-1.el8
```
I'm not sure why it downloads `rubygem-google-protobuf-doc`, that gem doesn't exists, and [google-protobuf](https://github.com/protocolbuffers/protobuf/blob/main/ruby/google-protobuf.gemspec) doesn't have it as dependency

**Comment by dgoetz** [comment]
It is not a gem, but a subpackage created because of the section `%package doc` in the spec file. This is normal and looking at the Koji task it is build properly. So just an explanation for the -doc, not the problem.

**Comment by stejskalleos** [comment]
> It is not a gem, but a subpackage created because of the section `%package doc` in the spec file. This is normal and looking at the Koji task it is build properly. So just an explanation for the -doc, not the problem.

Okay, how do I fix the rpm job? Should I remove the %doc section from spec file?

**Comment by ehelms** [comment]
The failure is a legit issue if you look at the provides on the RPM due to the Epoc:

```
$ rpm -qp rubygem-google-protobuf-3.21.6-1.el8.x86_64.rpm --provides
rubygem(google-protobuf) = 3.21.6
rubygem-google-protobuf = 1:3.21.6-1.el8
rubygem-google-protobuf(x86-64) = 1:3.21.6-1.el8
```

File: packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+BuildRequires: gcc
+# end specfile generated dependencies
+
+%description
+Protocol Buffers are Google data interchange format.
+
+
+%package doc
+Summary: Documentation for %{name}
+Requires: %{name} = %{version}-%{release}
```
Comment (ehelms) [inline packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec:29]:
```suggestion
Requires: %{name} = %{epoch}:%{version}-%{release}
```

File: packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# template: default
+%global gem_name google-protobuf
+%global gem_require_name %{gem_name}
+
+Name: rubygem-%{gem_name}
+Version: 3.21.6
+Release: 1%{?dist}
+Epoch: 1
```
Comment (ekohl) [inline packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec:8]: I may have missed this, but why is there an epoch?

Comment (ehelms) [reply packages/plugins/rubygem-google-protobuf/rubygem-google-protobuf.spec:8]: That's a great question, I cannot find a reason for having one as I do not see any competing RPMs that exist and we have not released one ourselves. I think then you can just drop this line and revert the changelog @stejskalleos

**Comment by ehelms** [comment]
I did some local build and testing and opened up https://github.com/theforeman/foreman-packaging/pull/8546

**Comment by ehelms** [comment]
Replaced by https://github.com/theforeman/foreman-packaging/pull/8546

**Pull request closed without merging**